src/java/org/apache/cassandra/serializers/UserTypeSerializer.java

```diff
@@ -36,9 +36,10 @@ public UserTypeSerializer(LinkedHashMap<String, TypeSerializer<?>> fields)
     public void validate(ByteBuffer bytes) throws MarshalException
     {
         ByteBuffer input = bytes.duplicate();
-        int i = 0;
+        int i = -1; // first thing in the loop is to increment, so when starting this will get set to 0 and match the fields
```

Comment:
This change doesn't seem necessary for the `MarshalException` check below, did you plan to assert something like at least one field exists?

Reply:
this was caused by `continue`.  If you have a null value then `continue` is called which didn't increment `I`, this caused a wrong message being sent to the user.

Reply:
Makes sense, thanks.